lib/_http_client.js

```diff
@@ -68,7 +68,11 @@ function ClientRequest(options, cb) {
   self.socketPath = options.socketPath;
   self.timeout = options.timeout;
 
-  var method = self.method = (options.method || 'GET').toUpperCase();
+  let method = options.method;
```

Comment:
Can you keep `var`. We haven't really made the switch to `let` in `lib/` code for performance reasons.